de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inventory/process/M_InventoryLine_AssignSelectedLines.java

```diff
@@ -0,0 +1,74 @@
+package de.metas.inventory.process;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+import org.adempiere.ad.dao.IQueryBL;
+import org.adempiere.ad.dao.IQueryBuilder;
+import org.adempiere.ad.dao.IQueryFilter;
+import org.adempiere.ad.trx.api.ITrx;
+import org.adempiere.util.Services;
+import org.compiere.model.I_M_InventoryLine;
+
+import de.metas.inventory.IInventoryBL;
+import de.metas.process.JavaProcess;
+import de.metas.process.Param;
+
+/*
+ * #%L
+ * de.metas.swat.base
+ * %%
+ * Copyright (C) 2018 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+public class M_InventoryLine_AssignSelectedLines extends JavaProcess
+{
+	private static final String PARAM_NumberOfCounters = "NumberOfCounters";
+
+	private final IInventoryBL inventoryBL = Services.get(IInventoryBL.class);
+
+	@Param(parameterName = PARAM_NumberOfCounters, mandatory = true)
+	private BigDecimal numberOfCounters;
+
+	@Override
+	protected String doIt() throws Exception
+	{
+		final List<I_M_InventoryLine> inventoryLines = getSelectedInventoryLines();
+
+		inventoryBL.assignToInventoryCounters(inventoryLines, numberOfCounters.intValue());
+
+		return MSG_OK;
+	}
+
+	private List<I_M_InventoryLine> getSelectedInventoryLines()
+	{
+		final IQueryFilter<I_M_InventoryLine> selectedPartners = getProcessInfo().getQueryFilter();
+
+		final IQueryBL queryBL = Services.get(IQueryBL.class);
+		final IQueryBuilder<I_M_InventoryLine> queryBuilder = queryBL.createQueryBuilder(I_M_InventoryLine.class, getCtx(), ITrx.TRXNAME_ThreadInherited);
```

Comment:
u don't have to provide the ctx and trx. use the createQueryBuilder(clazz) which is doing the same.

Reply:
fixed